File: app/src/main/java/com/kickstarter/ui/viewholders/discoverydrawer/LoggedOutViewHolder.java
package com.kickstarter.ui.viewholders.discoverydrawer;

import android.view.View;

import com.kickstarter.R;
import com.kickstarter.ui.viewholders.KSViewHolder;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import butterknife.ButterKnife;
import butterknife.OnClick;

public final class LoggedOutViewHolder extends KSViewHolder {
  private Delegate delegate;

  public interface Delegate {
    void loggedOutViewHolderInternalToolsClick(final @NonNull LoggedOutViewHolder viewHolder);
    void loggedOutViewHolderLoginToutClick(final @NonNull LoggedOutViewHolder viewHolder);
  }

  public LoggedOutViewHolder(final @NonNull View view, final @NonNull Delegate delegate) {
    super(view);
    ButterKnife.bind(this, view);
    this.delegate = delegate;
  }

  @Override
  public void bindData(final @Nullable Object data) throws Exception {
  }

  @Override
  public void onBind() {
  }

  @Nullable @OnClick(R.id.internal_tools_icon_button)
  public void internalToolsClick() {
    this.delegate.loggedOutViewHolderInternalToolsClick(this);
  }

  @OnClick(R.id.logged_out_container)
  public void loginToutClick() {
    this.delegate.loggedOutViewHolderLoginToutClick(this);
  }
}

File: app/src/main/java/com/kickstarter/ui/viewholders/discoverydrawer/LoggedOutViewHolder.kt
package com.kickstarter.ui.viewholders.discoverydrawer

import com.kickstarter.databinding.DiscoveryDrawerLoggedOutViewBinding
import com.kickstarter.ui.viewholders.KSViewHolder

class LoggedOutViewHolder(private val binding: DiscoveryDrawerLoggedOutViewBinding, private val delegate: Delegate) : KSViewHolder(binding.root) {
    interface Delegate {
        fun loggedOutViewHolderActivityClick(viewHolder: LoggedOutViewHolder)
        fun loggedOutViewHolderInternalToolsClick(viewHolder: LoggedOutViewHolder)
        fun loggedOutViewHolderLoginToutClick(viewHolder: LoggedOutViewHolder)
        fun loggedOutViewHolderHelpClick(viewHolder: LoggedOutViewHolder)
    }

    @Throws(Exception::class)
    override fun bindData(data: Any?) {
    }

    override fun onBind() {
        binding.drawerActivity.setOnClickListener {
            activityClick()
        }
        binding.drawerHelp.setOnClickListener {
            helpClick()
        }
        binding.internalToolsLayout.internalTools.setOnClickListener {
            internalToolsClick()
        }
        binding.loggedOutTextView.setOnClickListener {
            loginToutClick()
        }
    }

    private fun activityClick() {
        delegate.loggedOutViewHolderActivityClick(this)
    }

    private fun helpClick() {
        delegate.loggedOutViewHolderHelpClick(this)
    }

    private fun internalToolsClick() {
        delegate.loggedOutViewHolderInternalToolsClick(this)
    }

    private fun loginToutClick() {
        delegate.loggedOutViewHolderLoginToutClick(this)
    }
}
